_api-reference/index-apis/component-template.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: default
+title: Create or update component template
+parent: Index APIs
+nav_order: 29
+---
+
+# Create or update a component template
+
+You use the Create Component Template API to create or update a component template. A component template is a reusable building block that defines settings, mappings, and aliases that can be shared across multiple index templates. 
+
+An index template can be constructed from multiple component templates. To incorporate a component template into an index template, you need to list it in the `composed_of` section of the index template. Component templates are only applied to newly created data streams and indexes that match the criteria specified in the index template.
+
+If any settings or mappings are directly defined in the index template or the index creation request, those settings will take precedence over the settings or mappings specified in a component template.
+
+Component templates are utilized solely during the process of index creation. For data streams, this includes the creation of the data stream itself and the creation of the backing indexes that support the stream. Modifications made to component templates will not affect existing indexes, including the backing indexes of a data stream.
+
+## Path and HTTP methods
+
+```
+PUT _component_template/<component-template-name>
+GET _component_template/<component-template-name>
```

Comment:
should we mention this separately. It's not very clear that the below mentioned query parameters just belong to PUT request. Ideally GET request doesn't contain the body, but yeah if we can make this more clear

Reply:
To clarify at @sarthakaggarwal97, the PUT requests should contain both Query Parameters and a request body but the GET Request should not?

Reply:
Modified with the details given by @sarthakaggarwal97